docker/client.go

```diff
@@ -67,13 +70,61 @@ func CheckSocket(socketPath string) bool {
 	return true
 }
 
+// IsWithinHostNetwork checks if a provided target is within the host container network
+func IsWithinHostNetwork(socketPath string, targetAddress string, targetPort int) (bool, error) {
+	// Always enforce network validation
+	containers, err := ListContainers(socketPath, true)
+	if err != nil {
+
+		return false, err
```

Comment:
In go if err is nil you should use `return false, nil` specifically instead of `return false, err`